clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -750,19 +745,12 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                 shard, memref.instr.type != TRACE_TYPE_INSTR_CONDITIONAL_JUMP,
                 "The CONDITIONAL_JUMP type is deprecated and should not appear");
             if (!type_is_instr_direct_branch(memref.instr.type)) {
-                shard->last_indirect_target_ = shard->last_branch_marker_value_;
                 report_if_false(shard,
-                                shard->last_indirect_target_ != 0 &&
-                                    shard->prev_entry_.marker.type == TRACE_TYPE_MARKER &&
-                                    shard->prev_entry_.marker.marker_type ==
-                                        TRACE_MARKER_TYPE_BRANCH_TARGET,
-                                "Indirect branches must be preceded by their targets");
+                                // We assume the app doesn't actually target PC=0.
+                                memref.instr.indirect_branch_target != 0,
```

Comment:
Also assert that it's zero in all other cases.

Reply:
Done.